### ISSUE TITLE
[FEATURE REQUEST] Add adapter for mongodb

### ISSUE THREAD

**Issue by saniales** [tryvium-travels/golang-cache-adapters#1]
# MongoDB Adapter
The idea is based on using a collection as cache. The utility of this is to give an alternative in case the other caches are not available in a multi cache environment

TTL is handled using the `$ttl` field in mongo while the cache collection needs to have a hashed index on the `key` field for max performance

Differently from redis, a proper collection setup is required, just setting up the connection is not enough.

**Comment by saniales** [comment]
Struct proposal:

``` go
type CacheItem struct {
    data interface{}
    $ttl uint
}
```

**Comment by saniales** [comment]
Closed by #24
